File: OJ/PAT/basic_level/1023_组个最小数/min_permutation.py
from __future__ import print_function

us_num = raw_input().split()
us_num = map(int, us_num)

first_num = 1

for i in range(1, 10):
	if us_num[i]:
		first_num = i
		us_num[i] -= 1
		break

print(first_num, end='')
for i in range(10):
	while us_num[i]:
		print(i, end='')
		us_num[i] -= 1

File: OJ/PAT/basic_level/1023_组个最小数/min_permutation.c
#include <stdio.h>
#define MAX 11

int main(){
	int us_num[MAX];
	int first_num=1;
	int i=0;

	for(i=0; i<10; i++){
		scanf("%d", &us_num[i]);
	}

	for(i=1; i<10; i++){
		if( us_num[i] ){
			first_num = i;
			us_num[i]--;
			break;
		}
	}

	printf("%d", first_num);
	for(i=0; i<10; i++){
		while( us_num[i] ){
			printf("%i", i);
			us_num[i]--;
		}
	}

	return 0;
}
